src/ts/editorcanvas/CanvasVM.ts

```diff
@@ -133,15 +134,32 @@ module editorcanvas {
             this.systemLoader.loadMap(this.project.projectName, this.data.getEmptyClone())
                 .then((entitySystem : entityframework.EntitySystem) => {
                     this.changeData(entitySystem);
+                    this.loadAssets();
                 }, framework.error.onPromiseError(this._context));
         }
 
+        private loadAssets() {
+            this.numAssetsToLoad = this.data.assets.length;
+            this.data.assets.forEach((asset : entityframework.map.Asset) => {
```

Comment:
The editor should not treat the assets in the file as a source of truth.  It should instead only rely on the information in the entities.